homeassistant/components/recorder.py

```diff
@@ -475,6 +486,32 @@ def _close_run(self):
             "UPDATE recorder_runs SET end=? WHERE start=?",
             (dt_util.utcnow(), self.recording_start))
 
+    def _purge_old_data(self):
+        """Purge events and states older than purge_days ago."""
+        if not self.purge_days or self.purge_days < 1:
+            _LOGGER.debug("purge_days set to %s, will not purge any old data.",
```

Comment:
No need for this if you never call it when purge_days is not None, like i suggested above.